File: app/src/main/java/com/usher/demo/rx/RxSumActivity.java
package com.usher.demo.rx;

import android.os.Bundle;
import android.widget.Button;

import com.jakewharton.rxbinding3.view.RxView;
import com.usher.demo.R;
import com.usher.demo.base.BaseActivity;
import com.usher.demo.utils.RxUtil;

import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.Observable;

public class RxSumActivity extends BaseActivity {
    @BindView(R.id.param1_button)
    Button mParam1Button;
    @BindView(R.id.param2_button)
    Button mParam2Button;
    @BindView(R.id.sum_button)
    Button mSumButton;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rx_sum);
        ButterKnife.bind(this);

        initView();
    }

    private void initView() {
        Observable<Integer> parm1$ = RxView.clicks(mParam1Button)
                .map(v -> 1)
                .scan((acc, curr) -> acc + curr)
                .doOnNext(v -> mParam1Button.setText(String.valueOf(v)));

        Observable<Integer> parm2$ = RxView.clicks(mParam2Button)
                .map(v -> 1)
                .scan((acc, curr) -> acc + curr)
                .doOnNext(v -> mParam2Button.setText(String.valueOf(v)));

        Observable.combineLatest(
                parm1$,
                parm2$,
                (first, second) -> first + second
        )
                .as(RxUtil.autoDispose(this))
                .subscribe(v -> mSumButton.setText(String.valueOf(v)));
    }
}

File: app/src/main/java/com/usher/demo/rx/RxSumActivity.kt
package com.usher.demo.rx

import com.jakewharton.rxbinding4.view.clicks
import com.twigcodes.ui.util.RxUtil
import com.usher.demo.R
import com.usher.demo.base.BaseActivity
import io.reactivex.rxjava3.core.Observable
import kotlinx.android.synthetic.main.activity_rx_sum.*

class RxSumActivity : BaseActivity(R.layout.activity_rx_sum, Theme.LIGHT_AUTO) {

    override fun initView() {
        val param1 = param1_textview.clicks()
                .map { 1 }
                .scan { acc, value -> acc + value }
                .doOnNext { param1_textview.text = it.toString() }

        val param2 = param2_textview.clicks()
                .map { 1 }
                .scan { acc, value -> acc + value }
                .doOnNext { param2_textview.text = it.toString() }

        Observable.combineLatest(param1, param2) { t1, t2 -> t1 + t2 }
                .to(RxUtil.autoDispose(this))
                .subscribe { sum_textview.text = it.toString() }
    }
}